components/nimbus/src/nimbus_client.rs

```diff
@@ -325,7 +326,11 @@ impl NimbusClient {
                     is_enrolled_set.insert(ee.slug.clone());
                     all_enrolled_set.insert(ee.slug.clone());
                 }
-                EnrollmentStatus::WasEnrolled { .. } => {
+                EnrollmentStatus::WasEnrolled { .. }
+                | EnrollmentStatus::Disqualified {
+                    reason: DisqualifiedReason::NotSelected | DisqualifiedReason::NotTargeted,
+                    ..
+                } => {
```

Comment:
If `all_enrolled_set` is the set of experiments that the user is or was ever in an enrolled state, then this should include `OptOut` and `Error` disqualifications, no?
```suggestion
                | EnrollmentStatus::Disqualified { .. } => {
```

Reply:
Yeah that's fair, I'll adjust that

Reply:
Also with this I'm adding a section to the unit test where it reads from the database to fetch all the enrollments and verifies that each of the experiments are in the status we expect them to be in. 

The choice to do this is two-fold, it validates that the logic surrounding disqualification for various reasons continues to work as expected but more importantly it serves as documentation of what's being tested.